tss-esapi/src/structures/lists/command_code.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    constants::CommandCode,
+    tss2_esys::{TPM2_MAX_CAP_CC, TPML_CC},
+    Error, Result, WrapperErrorKind,
+};
+use log::error;
+use std::convert::TryFrom;
+
+/// A list of command codes.
+#[derive(Debug, Clone, Default)]
+pub struct CommandCodeList {
+    command_codes: Vec<CommandCode>,
+}
+
+impl CommandCodeList {
+    pub const MAX_SIZE: usize = TPM2_MAX_CAP_CC as usize;
+    /// Creates a new CommandCodeList
+    pub const fn new() -> Self {
+        CommandCodeList {
+            command_codes: Vec::new(),
+        }
+    }
+
+    /// Returns a command codes in the list as a slice.
+    pub fn value(&self) -> &[CommandCode] {
```

Comment:
Just wondering if we shouldn't also implement `AsRef<[CommandCode]>` ([`AsRef` docs](https://doc.rust-lang.org/std/convert/trait.AsRef.html)). This would make it consistent with all other Rust APIs that expose inner buffers (e.g. String exposes ref of bytes etc).

Thinking on this a little bit longer maybe there should be also a function to convert the CommandCodeList into raw `Vec<CommandCode>` just like [other Rust APIs using `into_inner`](https://users.rust-lang.org/t/explanation-of-into-inner/13872).

On more thing that could be useful is `impl TryFrom<Vec<CommandCode>> for CommandCodeList` that'd just check for the correct length and if that's fine wrap the vector with the CommandCodeList.

Thoughts?

Reply:
I see no problem with it what so ever. I will start working on it.

Reply:
fixed